### PR TITLE
Support auth credentials for private registry push

### LINKED ISSUE #35: Can't push image: Error: Status 401 trying to push repository...
I can't seem to push a docker image via the `mvn docker:push` goal. At the same time I am able to do so successfully via the regular `docker push` command

I use the following `pom.xml`: https://gist.github.com/mattijsf/c93be14f3680c9cb96b8

1. I run `mvn docker:build`
2. As a result the image `mattijsf/docker-test-project:latest` is created
3. I run `mvn docker:push -X`. I receive the following output
```bash
bash-4.3$ mvn docker:push -X
Apache Maven 3.2.1 (ea8b2b07643dbb1b84b6d16e1f08391b666bc1e9; 2014-02-14T18:37:52+01:00)
Maven home: /usr/local/Cellar/maven/3.2.1/libexec
Java version: 1.7.0_55, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_55.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.10", arch: "x86_64", family: "mac"
[INFO] Error stacktraces are turned on.
[DEBUG] Reading global settings from /usr/local/Cellar/maven/3.2.1/libexec/conf/settings.xml
[DEBUG] Reading user settings from /Users/mattijs/.m2/settings.xml
[DEBUG] Using local repository at /Users/mattijs/.m2/repository
[DEBUG] Using manager EnhancedLocalRepositoryManager with priority 10.0 for /Users/mattijs/.m2/repository
[INFO] Scanning for projects...
[DEBUG] Extension realms for project com.docker.test:docker-test-project:pom:0.0.1-SNAPSHOT: (none)
[DEBUG] Looking up lifecyle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Resolving plugin prefix docker from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix docker to com.spotify:docker-maven-plugin from POM com.docker.test:docker-test-project:pom:0.0.1-SNAPSHOT
[DEBUG] === REACTOR BUILD PLAN ================================================
[DEBUG] Project: com.docker.test:docker-test-project:pom:0.0.1-SNAPSHOT
[DEBUG] Tasks:   [docker:push]
[DEBUG] Style:   Regular
[DEBUG] =======================================================================
[INFO] 
[INFO] Using the builder org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder with a thread count of 1
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building docker-test-project 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[DEBUG] Resolving plugin prefix docker from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix docker to com.spotify:docker-maven-plugin from POM com.docker.test:docker-test-project:pom:0.0.1-SNAPSHOT
[DEBUG] Lifecycle default -> [validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy]
[DEBUG] Lifecycle clean -> [pre-clean, clean, post-clean]
[DEBUG] Lifecycle site -> [pre-site, site, post-site, site-deploy]
[DEBUG] === PROJECT BUILD PLAN ================================================
[DEBUG] Project:       com.docker.test:docker-test-project:0.0.1-SNAPSHOT
[DEBUG] Dependencies (collect): []
[DEBUG] Dependencies (resolve): []
[DEBUG] Repositories (dependencies): [central (http://repo.maven.apache.org/maven2, releases)]
[DEBUG] Repositories (plugins)     : [central (http://repo.maven.apache.org/maven2, releases)]
[DEBUG] -----------------------------------------------------------------------
[DEBUG] Goal:          com.spotify:docker-maven-plugin:0.0.23:push (default-cli)
[DEBUG] Style:         Regular
[DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <dockerHost>${dockerHost}</dockerHost>
  <imageName>mattijsf/docker-test-project</imageName>
  <execution default-value="${mojoExecution}"/>
  <session default-value="${session}"/>
</configuration>
[DEBUG] =======================================================================
[INFO] 
[INFO] --- docker-maven-plugin:0.0.23:push (default-cli) @ docker-test-project ---
[DEBUG] Created new class realm maven.api
[DEBUG] Importing foreign packages into class realm maven.api
[DEBUG]   Imported: org.apache.maven.wagon.events < plexus.core
[DEBUG]   Imported: org.eclipse.aether.impl < plexus.core
[DEBUG]   Imported: org.apache.maven.exception < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.Xpp3Dom < plexus.core
[DEBUG]   Imported: org.eclipse.aether.version < plexus.core
[DEBUG]   Imported: org.eclipse.aether.metadata < plexus.core
[DEBUG]   Imported: org.eclipse.aether.collection < plexus.core
[DEBUG]   Imported: org.apache.maven.monitor < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.resource < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.logging < plexus.core
[DEBUG]   Imported: org.apache.maven.profiles < plexus.core
[DEBUG]   Imported: org.apache.maven.classrealm < plexus.core
[DEBUG]   Imported: org.apache.maven.execution.scope < plexus.core
[DEBUG]   Imported: org.eclipse.aether.artifact < plexus.core
[DEBUG]   Imported: org.apache.maven.execution < plexus.core
[DEBUG]   Imported: org.apache.maven.reporting < plexus.core
[DEBUG]   Imported: org.apache.maven.usability < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.container < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.component < plexus.core
[DEBUG]   Imported: org.eclipse.aether.transfer < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authentication < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlSerializer < plexus.core
[DEBUG]   Imported: org.apache.maven.lifecycle < plexus.core
[DEBUG]   Imported: org.eclipse.aether.* < plexus.core
[DEBUG]   Imported: org.eclipse.aether.graph < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.classworlds < plexus.core
[DEBUG]   Imported: org.eclipse.aether.internal.impl < plexus.core
[DEBUG]   Imported: org.eclipse.aether.repository < plexus.core
[DEBUG]   Imported: org.eclipse.aether.resolution < plexus.core
[DEBUG]   Imported: javax.inject.* < plexus.core
[DEBUG]   Imported: org.apache.maven.settings < plexus.core
[DEBUG]   Imported: org.codehaus.classworlds < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.* < plexus.core
[DEBUG]   Imported: org.apache.maven.toolchain < plexus.core
[DEBUG]   Imported: org.eclipse.aether.spi < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.observers < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParserException < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParser < plexus.core
[DEBUG]   Imported: org.apache.maven.configuration < plexus.core
[DEBUG]   Imported: org.apache.maven.cli < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.context < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authorization < plexus.core
[DEBUG]   Imported: org.apache.maven.project < plexus.core
[DEBUG]   Imported: org.eclipse.aether.installation < plexus.core
[DEBUG]   Imported: org.eclipse.aether.deployment < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.lifecycle < plexus.core
[DEBUG]   Imported: org.apache.maven.rtinfo < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.configuration < plexus.core
[DEBUG]   Imported: org.apache.maven.artifact < plexus.core
[DEBUG]   Imported: org.apache.maven.model < plexus.core
[DEBUG]   Imported: org.slf4j.* < plexus.core
[DEBUG]   Imported: javax.enterprise.inject.* < plexus.core
[DEBUG]   Imported: org.apache.maven.* < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.proxy < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.* < plexus.core
[DEBUG]   Imported: org.apache.maven.plugin < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.personality < plexus.core
[DEBUG] Populating class realm maven.api
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=1, ConflictMarker.markTime=0, ConflictMarker.nodeCount=102, ConflictIdSorter.graphTime=1, ConflictIdSorter.topsortTime=0, ConflictIdSorter.conflictIdCount=57, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=17, ConflictResolver.conflictItemCount=97, DefaultDependencyCollector.collectTime=512, DefaultDependencyCollector.transformTime=21}
[DEBUG] com.spotify:docker-maven-plugin:jar:0.0.23:
[DEBUG]    com.spotify:docker-client:jar:2.6.4:compile
[DEBUG]       org.slf4j:slf4j-api:jar:1.7.6:compile
[DEBUG]       com.google.guava:guava:jar:12.0.1:compile
[DEBUG]          com.google.code.findbugs:jsr305:jar:1.3.9:compile
[DEBUG]       org.apache.commons:commons-compress:jar:1.8.1:compile
[DEBUG]       org.apache.httpcomponents:httpclient:jar:4.3.5:compile
[DEBUG]          org.apache.httpcomponents:httpcore:jar:4.3.2:compile
[DEBUG]          commons-logging:commons-logging:jar:1.1.3:compile
[DEBUG]          commons-codec:commons-codec:jar:1.6:compile
[DEBUG]       de.gesellix:unix-socket-factory:jar:2014-09-26T18-52-33:compile
[DEBUG]          org.codehaus.groovy:groovy-all:jar:2.3.6:runtime
[DEBUG]          org.codehaus.groovy.modules.http-builder:http-builder:jar:0.7.1:runtime
[DEBUG]             net.sf.json-lib:json-lib:jar:jdk15:2.3:runtime
[DEBUG]                commons-beanutils:commons-beanutils:jar:1.8.0:runtime
[DEBUG]                commons-collections:commons-collections:jar:3.2.1:runtime
[DEBUG]                commons-lang:commons-lang:jar:2.4:runtime
[DEBUG]                net.sf.ezmorph:ezmorph:jar:1.0.6:runtime
[DEBUG]             net.sourceforge.nekohtml:nekohtml:jar:1.9.16:runtime
[DEBUG]                xerces:xercesImpl:jar:2.9.1:runtime
[DEBUG]                   xml-apis:xml-apis:jar:1.3.04:runtime
[DEBUG]             xml-resolver:xml-resolver:jar:1.2:runtime
[DEBUG]       org.bouncycastle:bcpkix-jdk15on:jar:1.51:compile
[DEBUG]          org.bouncycastle:bcprov-jdk15on:jar:1.51:compile
[DEBUG]    com.typesafe:config:jar:1.2.0:compile
[DEBUG]    org.apache.maven:maven-core:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-settings:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-settings-builder:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-repository-metadata:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-plugin-api:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-model-builder:jar:3.2.1:compile
[DEBUG]       org.apache.maven:maven-aether-provider:jar:3.2.1:compile
[DEBUG]          org.eclipse.aether:aether-spi:jar:0.9.0.M2:compile
[DEBUG]       org.eclipse.aether:aether-impl:jar:0.9.0.M2:compile
[DEBUG]       org.eclipse.aether:aether-api:jar:0.9.0.M2:compile
[DEBUG]       org.eclipse.aether:aether-util:jar:0.9.0.M2:compile
[DEBUG]       org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.0.0.M5:compile
[DEBUG]          javax.enterprise:cdi-api:jar:1.0:compile
[DEBUG]             javax.annotation:jsr250-api:jar:1.0:compile
[DEBUG]             javax.inject:javax.inject:jar:1:compile
[DEBUG]          org.sonatype.sisu:sisu-guice:jar:no_aop:3.1.0:compile
[DEBUG]             aopalliance:aopalliance:jar:1.0:compile
[DEBUG]          org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.0.0.M5:compile
[DEBUG]       org.codehaus.plexus:plexus-interpolation:jar:1.19:compile
[DEBUG]       org.codehaus.plexus:plexus-utils:jar:3.0.17:compile
[DEBUG]       org.codehaus.plexus:plexus-classworlds:jar:2.5.1:compile
[DEBUG]       org.codehaus.plexus:plexus-component-annotations:jar:1.5.5:compile
[DEBUG]       org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile
[DEBUG]          org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]    org.apache.maven:maven-artifact:jar:3.2.1:compile
[DEBUG]    org.eclipse.jgit:org.eclipse.jgit:jar:3.2.0.201312181205-r:compile
[DEBUG]       com.jcraft:jsch:jar:0.1.46:compile
[DEBUG]       com.googlecode.javaewah:JavaEWAH:jar:0.5.6:compile
[DEBUG]    com.fasterxml.jackson.core:jackson-databind:jar:2.2.3:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-annotations:jar:2.2.3:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-core:jar:2.2.3:compile
[DEBUG] Created new class realm plugin>com.spotify:docker-maven-plugin:0.0.23
[DEBUG] Importing foreign packages into class realm plugin>com.spotify:docker-maven-plugin:0.0.23
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>com.spotify:docker-maven-plugin:0.0.23
[DEBUG]   Included: com.spotify:docker-maven-plugin:jar:0.0.23
[DEBUG]   Included: com.spotify:docker-client:jar:2.6.4
[DEBUG]   Included: org.slf4j:slf4j-api:jar:1.7.6
[DEBUG]   Included: com.google.guava:guava:jar:12.0.1
[DEBUG]   Included: com.google.code.findbugs:jsr305:jar:1.3.9
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.8.1
[DEBUG]   Included: org.apache.httpcomponents:httpclient:jar:4.3.5
[DEBUG]   Included: org.apache.httpcomponents:httpcore:jar:4.3.2
[DEBUG]   Included: commons-logging:commons-logging:jar:1.1.3
[DEBUG]   Included: commons-codec:commons-codec:jar:1.6
[DEBUG]   Included: de.gesellix:unix-socket-factory:jar:2014-09-26T18-52-33
[DEBUG]   Included: org.codehaus.groovy:groovy-all:jar:2.3.6
[DEBUG]   Included: org.codehaus.groovy.modules.http-builder:http-builder:jar:0.7.1
[DEBUG]   Included: net.sf.json-lib:json-lib:jar:jdk15:2.3
[DEBUG]   Included: commons-beanutils:commons-beanutils:jar:1.8.0
[DEBUG]   Included: commons-collections:commons-collections:jar:3.2.1
[DEBUG]   Included: commons-lang:commons-lang:jar:2.4
[DEBUG]   Included: net.sf.ezmorph:ezmorph:jar:1.0.6
[DEBUG]   Included: net.sourceforge.nekohtml:nekohtml:jar:1.9.16
[DEBUG]   Included: xerces:xercesImpl:jar:2.9.1
[DEBUG]   Included: xml-apis:xml-apis:jar:1.3.04
[DEBUG]   Included: xml-resolver:xml-resolver:jar:1.2
[DEBUG]   Included: org.bouncycastle:bcpkix-jdk15on:jar:1.51
[DEBUG]   Included: org.bouncycastle:bcprov-jdk15on:jar:1.51
[DEBUG]   Included: com.typesafe:config:jar:1.2.0
[DEBUG]   Included: org.eclipse.aether:aether-util:jar:0.9.0.M2
[DEBUG]   Included: javax.enterprise:cdi-api:jar:1.0
[DEBUG]   Included: javax.annotation:jsr250-api:jar:1.0
[DEBUG]   Included: javax.inject:javax.inject:jar:1
[DEBUG]   Included: org.sonatype.sisu:sisu-guice:jar:no_aop:3.1.0
[DEBUG]   Included: aopalliance:aopalliance:jar:1.0
[DEBUG]   Included: org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.0.0.M5
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.19
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.0.17
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.5.5
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.eclipse.jgit:org.eclipse.jgit:jar:3.2.0.201312181205-r
[DEBUG]   Included: com.jcraft:jsch:jar:0.1.46
[DEBUG]   Included: com.googlecode.javaewah:JavaEWAH:jar:0.5.6
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-databind:jar:2.2.3
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-annotations:jar:2.2.3
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-core:jar:2.2.3
[DEBUG]   Excluded: org.apache.maven:maven-core:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-model:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-settings:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-settings-builder:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-repository-metadata:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-plugin-api:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-model-builder:jar:3.2.1
[DEBUG]   Excluded: org.apache.maven:maven-aether-provider:jar:3.2.1
[DEBUG]   Excluded: org.eclipse.aether:aether-spi:jar:0.9.0.M2
[DEBUG]   Excluded: org.eclipse.aether:aether-impl:jar:0.9.0.M2
[DEBUG]   Excluded: org.eclipse.aether:aether-api:jar:0.9.0.M2
[DEBUG]   Excluded: org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.0.0.M5
[DEBUG]   Excluded: org.codehaus.plexus:plexus-classworlds:jar:2.5.1
[DEBUG]   Excluded: org.apache.maven:maven-artifact:jar:3.2.1
[DEBUG] Configuring mojo com.spotify:docker-maven-plugin:0.0.23:push from plugin realm ClassRealm[plugin>com.spotify:docker-maven-plugin:0.0.23, parent: sun.misc.Launcher$AppClassLoader@77fe0d66]
[DEBUG] Configuring mojo 'com.spotify:docker-maven-plugin:0.0.23:push' with basic configurator -->
[DEBUG]   (f) imageName = mattijsf/docker-test-project
[DEBUG]   (f) execution = com.spotify:docker-maven-plugin:0.0.23:push {execution: default-cli}
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@44951097
[DEBUG] -- end configuration --
[INFO] Pushing mattijsf/docker-test-project
The push refers to a repository [mattijsf/docker-test-project] (len: 1)
Sending image list
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.451 s
[INFO] Finished at: 2014-11-09T14:52:49+01:00
[INFO] Final Memory: 17M/156M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.0.23:push (default-cli) on project docker-test-project: Exception caught: Error: Status 401 trying to push repository mattijsf/docker-test-project: "" -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.spotify:docker-maven-plugin:0.0.23:push (default-cli) on project docker-test-project: Exception caught
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
  at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:108)
  at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
  at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
  at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:116)
  at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:361)
  at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
  at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
  at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
  at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:606)
  at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
  at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
  at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
  at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception caught
  at com.spotify.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:63)
  at com.spotify.docker.PushMojo.execute(PushMojo.java:38)
  at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
  at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
  ... 19 more
Caused by: com.spotify.docker.client.DockerException: Error: Status 401 trying to push repository mattijsf/docker-test-project: ""
  at com.spotify.docker.client.AnsiProgressHandler.progress(AnsiProgressHandler.java:55)
  at com.spotify.docker.client.ProgressStream.tail(ProgressStream.java:78)
  at com.spotify.docker.client.DefaultDockerClient.push(DefaultDockerClient.java:561)
  at com.spotify.docker.Utils.pushImage(Utils.java:59)
  at com.spotify.docker.PushMojo.execute(PushMojo.java:47)
  at com.spotify.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:61)
  ... 22 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
bash-4.3$ 
```
When I use the command `docker push mattijsf/docker-test-project` it is successfully able to upload the image. I also tried to run `docker login` again but I didn't have any impact. 

Just as a sidenote I also saw this exact same error before updating to the latest docker / boot2docker for which HTTPS/TLS is "required". So I don't expect that this is related.

Since 401 is unauthorized I was wondering if I am missing a way to provide the credentials to the docker-maven-plugin... ?

### PR DESCRIPTION
**Pull request by davidxia**
Fixes #35

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,3 +111,32 @@ will need to do this binding so the image gets built when maven is run from the
...
+credentials in your Maven's global `settings.xml` file. `docker.registry.server-address` is
+optional and defaults to `https://index.docker.io/v1/`.
+
+    <settings>
+      [...]
+      <profiles>
+        <profile>
+          <id>private-docker-registry</id>
+          <properties>
+            <docker.registry.username>foo</docker.registry.username>
```
Comment (davidxia) [inline README.md]: Are there maven property naming conventions?

**Comment by davidxia** [comment]
@Kobolog @rculbertson

**Comment by davidxia** [comment]
@Kobolog @rculbertson want to take a look at this when you get a chance?

File: src/main/java/com/spotify/docker/AbstractDockerMojo.java [hunk 2]
```diff
@@ -59,6 +72,22 @@ public void execute() throws MojoExecutionException {
...
+        authConfigBuilder.email(dockerRegistryEmail);
+      }
+      if (!isNullOrEmpty(dockerRegistryPassword)) {
+        authConfigBuilder.password(dockerRegistryPassword);
+      }
+      if (!isNullOrEmpty(dockerRegistryServerAddress)) {
+        authConfigBuilder.serverAddress(dockerRegistryServerAddress);
+      }
+
+      builder.authConfig(authConfigBuilder.build());
```
Comment (Kobolog) [inline src/main/java/com/spotify/docker/AbstractDockerMojo.java]: I'd say some combinations here don't make sense, like password without username, for example. Should it complain when such a combination is detected?

File: README.md (hunk 1 above)
Comment (Kobolog) [inline README.md]: No idea =)

**Comment by filosganga** [comment]
Hi, I am very keen to see this feature rolled out. Is there anything I can help to push forward on this?

File: src/main/java/com/spotify/docker/AbstractDockerMojo.java (hunk 2 above)
Comment (davidxia) [inline src/main/java/com/spotify/docker/AbstractDockerMojo.java]: Yea i can add a log message and just not add the auth creds?

**Comment by davidxia** [comment]
@rohansingh In case you want to review as well.

File: README.md (hunk 1 above)
Comment (bspeakmon) [inline README.md]: Username/password info is usually part of a `<servers>/<server>` block in `settings.xml`, not a profile.

**Comment by rohansingh** [comment]
:+1:

**Comment by davidxia** [comment]
closing in favor of https://github.com/spotify/docker-maven-plugin/pull/60

**Pull request closed without merging**